test(nx-plugin-e2e): refactor to use new test helper

BioPhoton opened the pull request:
Precondition of #773 

Included changes:
- use new test helper
- fix package name
- add skip flags to enable more tests
- test skip flags

matejchalk commented on e2e/nx-plugin-e2e/tests/configuration.e2e.test.ts:
```diff
@@ -1,35 +1,138 @@
+    const cwd = join(baseDir, 'configure');
+    await materializeTree(tree, cwd);
+
+    const { stderr } = await executeConfigurationGenerator(
```
These tests just became way less readable :disappointed: Before it was simple and direct, now everything's extracted to wrapper functions and constants.

For example, previously the test simply ran `executeProcess` with the CLI command, now it's "abstracted" away in a `executeConfigurationGenerator` via another  `executeGenerator` wrapper. This obfuscates the _actual user interaction_ which is being tested.

The _user_ runs the `nx g @code-pushup/nx-plugin:configuration ...` commands and sees the `CREATE ... code-pushup.config.ts` logs, so extracting those things out of the tests is counter-productive. When I read an `it`-block, I want to understand what's happening without having to click through a bunch of testing "abstractions" and extracted setup code. 

It's like instead of writing a simple statement like `console.log('Hello, world!')`, you break it down and spread it around as if to hide how simple it could be :sweat_smile: 

```ts
// pointlessly extracted constant
import { NAME } from './some-other-file-you-have-to-know-about'

// ... code to scroll through ...

// pointless wrapper function
function msg(prefix: string, suffix: string = '!') {
  return `${prefix}${NAME}${suffix}`;
}

// ... more code to scroll through ...

// another pointless wrapper
const log = () => {
  console.log(msg('Hello ', '!'));
}

// ... more code to scroll through ...

// "It's slightly shorter, that means the code is better, right?"
log();
```

BioPhoton replied:
refactored

matejchalk replied:
Thanks for the refactor, now I can understand each test by reading it top to bottom :slightly_smiling_face:

matejchalk commented on e2e/nx-plugin-e2e/tests/configuration.e2e.test.ts:
```diff
@@ -1,35 +1,138 @@
+    );
+  });
+
+  it('should generate conde-pushup.config.ts file and add target to project.json', async () => {
```
```suggestion
  it('should generate code-pushup.config.ts file and add target to project.json', async () => {
```

matejchalk commented on e2e/nx-plugin-e2e/tests/configuration.e2e.test.ts:
```diff
@@ -1,35 +1,138 @@
+    );
+  });
+
+  it('should generate conde-pushup.config.ts file and add target to project.json', async () => {
```
I don't see how you're asserting that the target was added.

BioPhoton replied:
As in mentioned here:
https://github.com/code-pushup/cli/pull/785#discussion_r1705579457

matejchalk commented on e2e/nx-plugin-e2e/tests/configuration.e2e.test.ts:
```diff
@@ -1,35 +1,138 @@
+    expect(cleanedStdout).toMatch(updateProjectMsgRegex);
+  });
+
+  it('should NOT conde-pushup.config.ts file if one already exists', async () => {
```
```suggestion
  it('should NOT create a code-pushup.config.ts file if one already exists', async () => {
```

matejchalk commented on e2e/nx-plugin-e2e/tests/configuration.e2e.test.ts:
```diff
@@ -1,35 +1,138 @@
+    expect(cleanedStdout).toMatch(updateProjectMsgRegex);
+  });
+
+  it('should NOT create conde-pushup.config.ts file if skipConfig is given', async () => {
```
```suggestion
  it('should NOT create a code-pushup.config.ts file if skipConfig is given', async () => {
```

matejchalk commented on e2e/nx-plugin-e2e/tests/configuration.e2e.test.ts:
```diff
@@ -1,35 +1,138 @@
+    expect(cleanedStdout).toMatch(updateProjectMsgRegex);
+  });
+
+  it('should NOT add target to project.json if skipTarget is given', async () => {
```
How are you testing that the target wasn't added? There's assertion for that in the test.

BioPhoton replied:
As we use dryRun i test it over the comments as all other checks in this file.

`expect(cleanedStdout).not.toMatch(/^UPDATE.*project.json/m);`

If the package would have been added I would see this msg

matejchalk commented on testing/test-nx-utils/src/lib/utils/tree.unit.test.ts:
Was this file committed by mistake? This file was renamed in https://github.com/code-pushup/cli/pull/777#discussion_r1703845862.

BioPhoton replied:
ups

BioPhoton commented on e2e/nx-plugin-e2e/mocks/utils.ts:
```diff
@@ -0,0 +1,28 @@
+import {join, relative} from "node:path";
+import {executeProcess} from "@code-pushup/utils";
```
formatted. wondering how this could happen... 🤔

BioPhoton commented on e2e/nx-plugin-e2e/mocks/utils.ts:
```diff
@@ -0,0 +1,28 @@
+}
+export function pluginFilePath(cwd: string): string {
+  return relative(join(process.cwd(), cwd), join(process.cwd(), "packages/nx-plugin/src/index.ts"));
+}
```
renamed

BioPhoton commented on e2e/nx-plugin-e2e/vite.config.e2e.ts:
```diff
@@ -17,6 +17,9 @@ export default defineConfig({
-    setupFiles: ['../../testing/test-setup/src/lib/reset.mocks.ts'],
+    setupFiles: [
+      '../../testing/test-setup/src/lib/reset.mocks.ts',
+      '../../testing/test-setup/src/lib/console.mock.ts',
```
removed

BioPhoton commented:
I now added tests to check the file system.

The reason why I did not do that i previous PR's was because our generator also tried to install the packages and our e2e setup does not allow this ATM.

So now i added a flag to skip the installation of the packages. I also added tests for the file system now as it can avoid errors with `--skipInstall`.

Have a look.
